mirdata/datasets/haydn_op20.py

```diff
@@ -0,0 +1,310 @@
+        filename="haydnop20v1.3_annotated.zip",
+        url="https://github.com/napulen/haydn_op20_harm/releases/download/v1.3/haydnop20v1.3_annotated.zip",
+        checksum="1c65c8da312e1c9dda681d0496bf527f",
+        destination_dir=".",
+    )
+}
+DOWNLOAD_INFO = ""
+
+LICENSE_INFO = (
+    "Creative Commons Attribution Non Commercial Share Alike 4.0 International."
+)
+
+
+class Track(core.Track):
+    """haydn op20 track class
+
+    Args:
+        track_id (str): track id of the track
+
+    Attributes:
+        title (str): title of the track
+        track_id (str): track id
+        humdrum_annotated_path (str): path to humdrum annotated score
+
+    Cached Properties:
+        key (str): key annotation
+        score (np.array): computed audio spectrum
+        keys (list): annotated local key
+        roman_numerals (list): MusicBrainz metadata
+        chords (list): annotated local key
+        duration (int): relative duration
+        midi_path (str): path to midi
+        score (music21.stream.Score): music21 score
+    """
+
+    def __init__(
+        self,
+        track_id,
+        data_home,
+        dataset_name,
+        index,
+        metadata,
+    ):
+        super().__init__(
+            track_id,
+            data_home,
+            dataset_name,
+            index,
+            metadata,
+        )
+        self.humdrum_annotated_path = os.path.join(
+            self._data_home, self._track_paths["annotations"][0]
+        )
+        self.title = os.path.splitext(self._track_paths["annotations"][0])[0]
+
+    @core.cached_property
+    def score(self) -> music21.stream.Score:
+        return load_score(self.humdrum_annotated_path)
+
+    @core.cached_property
+    def keys(self, resolution=28) -> List[dict]:
+        return load_key(self.humdrum_annotated_path, resolution)
+
+    @core.cached_property
+    def roman_numerals(self, resolution=28) -> List[dict]:
+        return load_roman_numerals(self.humdrum_annotated_path, resolution)
+
+    @core.cached_property
+    def chords(self, resolution=28) -> List[dict]:
+        return load_roman_numerals(self.humdrum_annotated_path, resolution)
+
+    @core.cached_property
+    def duration(self) -> int:
+        return self.chords[-1]["time"]
+
+    @core.cached_property
+    def midi_path(self) -> str:
+        return load_midi_path(self.humdrum_annotated_path)
+
+    def to_jams(self):
+        """Get the track's data in jams format
+
+        Returns:
+            jams.JAMS: the track's data in jams format
+
+        """
+        return jams_utils.jams_converter(
+            metadata={
+                "duration": self.duration,
+                "title": self.title,
+                "key": self.keys,
+                "chords": self.chords,
+                "roman_numerals": self.roman_numerals,
+                "midi_path": self.midi_path,
+                "humdrum_annotated_path": self.humdrum_annotated_path,
+            },
+        )
+
+
```

Comment:
you can add use the `@io.coerce_to_string_io` decorator and do `load_chords(fhandle: TextIO)` to make all the loaders flexible, see details [here](https://mirdata--447.org.readthedocs.build/en/447/source/contributing.html#coerce-to-bytes-io-coerce-to-string-io). Same applies to the rest of the `load` functions with only `path` as parameter

Reply:
Music21 objects can be coerced too??

Reply:
everything can be converted to text file inst it?

Reply:
ok done